### PR TITLE
Add migration notes for 0.9 => 1.0

### PR DESCRIPTION
**Pull request by dfreeman**
I'll plan to link to this page from the `1.0.0-beta.1` release note. We can update with any further changes as needed during the beta cycle, and then I'll drop the beta disclaimer when we release `1.0.0` stable!

### CODE REVIEW

**Review by chriskrycho — APPROVED** [review]
Looks good! 👍🏼

File: docs/migrating.md [hunk 1]
```diff
@@ -1,5 +1,82 @@
 # Migrating
 
+## Glint 0.9.x to 1.0
+
+{% hint style="warning" %}
```
Comment (jamescdavis) [inline docs/migrating.md:5]: Is this supposed to render when [viewing the file](https://github.com/typed-ember/glint/blob/b0d5bf141a6ac202fb7a287bf2ee6a0d5ab065a0/docs/migrating.md)?

Comment (jamescdavis) [reply docs/migrating.md:5]: Oh wait, I forgot this gets pushed to GitBook. Looks good there.

File: docs/migrating.md [hunk 2]
```diff
@@ -1,5 +1,82 @@
...
+should likewise declare a `peerDependency` on `@glint/template` and **not** a regular `dependency`.
+
+### `include`/`exclude` Configuration
+
+Glint 1.0 drops support for the `transform` configuration key, which is where `include` and
+`exclude` globs were previously specified. These options were a blunt instrument left over from an
+earlier iteration of Glint.
+
+If you're currently relying on these keys to have Glint skip typechecking for parts of your
+codebase, consider using `@glint-nocheck` directives instead. You can automate the process of
```
Comment (jamescdavis) [inline docs/migrating.md:42]:
```suggestion
codebase, consider using [`@glint-nocheck`][glint-nocheck] directives instead. You can automate the process of
```

```md
[glint-nocheck]: directives.md#glint-nocheck
```
(or however you make internal links with fragments work 🙂 )

**Review by jamescdavis — APPROVED** [review]
One suggestion, but looks great!

**Pull request merged**